### PR TITLE
Implement new LabelEncoder in opset 2 in ML domain

### PR DESCRIPTION
**Pull request by wschin**
Based on my study, the current ONNXRuntime doesn't support [a new LabelEncoder](https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#aionnxmllabelencoder) introduced since previous ONNX release. This PR provides one implementation of it and also adds some tests. This operator is needed in ML.NET-to-ONNX conversion.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/ml/label_encoder.h [hunk 1]
```diff
@@ -43,5 +43,72 @@ class LabelEncoder final : public OpKernel {
...
+
+		auto input = X.template DataAsSpan<TKey>();
+		auto output = Y.template MutableDataAsSpan<TValue>();
+
+		for (int64_t i = 0; i < shape.Size(); ++i) {
+			if (_map.find(input[i]) == _map.end()) {
+				output[i] = _default_value;
+				continue;
+			}
+			output[i] = _map.at(input[i]);
```
Comment (jignparm) [inline onnxruntime/core/providers/cpu/ml/label_encoder.h]: Simplify -- use if{} else{} ... don't need `continue` statement.

File: onnxruntime/test/providers/cpu/ml/label_encoder_test.cc [hunk 2]
```diff
@@ -42,5 +42,131 @@ TEST(LabelEncoder, IntToString) {
...
+  test.AddAttribute("keys_int64s", keys);
+  test.AddAttribute("values_floats", values);
+  test.AddAttribute("default_float", 55.66f);
+
+  test.AddInput<std::int64_t>("X", dims, input);
+  test.AddOutput<float>("Y", dims, output);
+
+  test.Run();
+}
+
```
Comment (jignparm) [inline onnxruntime/test/providers/cpu/ml/label_encoder_test.cc]: Do we need a test to catch some failing conditions-- i.e. if both **keys_strings** and **keys_floats** attributes are set? Users are likely to configure an operator instance with this type of mistake.

File: onnxruntime/core/providers/cpu/ml/label_encoder.h (hunk 1 above)
Comment (wschin) [reply onnxruntime/core/providers/cpu/ml/label_encoder.h]: Ok.

File: onnxruntime/test/providers/cpu/ml/label_encoder_test.cc (hunk 2 above)
Comment (wschin) [reply onnxruntime/test/providers/cpu/ml/label_encoder_test.cc]: It has been checked in ONNX's shape inference function.

File: onnxruntime/core/providers/cpu/ml/label_encoder.h [hunk 3]
```diff
@@ -43,5 +43,66 @@ class LabelEncoder final : public OpKernel {
...
+    if (tensor_pointer == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+    const Tensor& X = *tensor_pointer;
+    const TensorShape& shape = X.Shape();
+    Tensor& Y = *context->Output(0, TensorShape(shape));
+
+    auto input = X.template DataAsSpan<TKey>();
+    auto output = Y.template MutableDataAsSpan<TValue>();
+
+    for (int64_t i = 0; i < shape.Size(); ++i) {
+      if (_map.find(input[i]) == _map.end())
```
Comment (xadupre) [inline onnxruntime/core/providers/cpu/ml/label_encoder.h]: maybe reuse the result of find to avoid looking again in the map with ``.at(…)``

**Pull request merged**